### ISSUE TITLE
support raspberry pi?

### ISSUE THREAD

**Issue by mrMiimo** [nanosonde/sip2rtsp#1]
very interesting project... I'd like to try it, I think it's what I'm looking for.
I'm trying to install it but I think it doesn't support the arm7 architecture, the one used by pi, right?

**Comment by nanosonde** [comment]
Hi!
Yes, on the RPi 32-bit arm7 would be required.
If you manage to build this image for armv7 then it should work on the RPi as well:
https://github.com/nanosonde/pulsegst/blob/main/Makefile#L18

However, gstreamer has some problems compiling ffmpeg on armv7 with multiarch docker build on AMD64.

**Comment by mrMiimo** [comment]
oh i see, thanks i will take a look.
tnks 👍🏻

**Comment by nanosonde** [comment]
I have bumped the gstreamer version to 1.22.1, but the included ffmpeg stuff in gstreamer fails to build on armv7 arch.
Maybe it could help if we install the ffmpeg version with all libav* libs including the libav*-dev version so that gstreamer could use the existing distro libs on armv7.

Another way could be to use cerbero to build armv7 binaries.

A third way would be just to wait until gstreamer 1.22.x will be part of Debian or Ubuntu.

**Comment by nanosonde** [comment]
@mrMiimo 

Yesterday, I had a quick look into Debian Testing (=bookworm). It seems that they already included the latest stable release of gstreamer and ffmpeg for all architectures.

So I tried the third-way and removed the dependency to the custom build versions of gstreamer and pulseaudio (=pulsegst image). Instead I try to use the distro provided binaries.

The current status is this: https://github.com/nanosonde/sip2rtsp/pull/5

Basically, it seems to work. However, I have some stuttering audio, when the SIP call is established. I have to investigate further...

**Comment by mrMiimo** [comment]
just this morning i was trying to install it. I confess that I had little hope in the third option (at least in the short term).

I'll try again later by removing the dependency.
One question, if I already had a Baresip instance configured and active, could I use that in some way?

**Comment by nanosonde** [comment]
> One question, if I already had a Baresip instance configured and active, could I use that in some way?

Yes, but you have to to do some modifications.
First, you have to make sure that the gstreamer pulsesink and pulsesrc elements are able to access your pulseaudio daemon. Have a look at the diagram that I have drawn. Then make sure that your baresip instance also uses this pulseaudio daemon as described. My python app contacts the baresip instance via its JSON-based TCP control protocol. Just do not use localhost, but the real IP address of the baresip host.

You have to figure out the rest yourself as I am not going to support non-Docker issues.
As all of this is mostly existing well-known software which just has to be configured in a certain way plus my python app which controls the main aspect, I presume that it is hard to maintain this in a non-Docker env.

With the docker container we have a reproducible system.

**Comment by mrMiimo** [comment]
crystal clear explanation.
my requirement is due to the fact that i have a fully closed doorbell and i only switch from that baresip configuration.
I'm trying different ways, thanks.

**Comment by nanosonde** [comment]
@mrMiimo 
Just tried this branch: debian_bookworm_no_custom_pulseaudio
For me it works and it also compiles for armv7 and arm64.

**Comment by nanosonde** [comment]
Prebuilt docker images can be found here: https://github.com/nanosonde/sip2rtsp/pkgs/container/sip2rtsp
Click on OS/Arch tab and choose the arm/v7 image for example:

`docker pull ghcr.io/nanosonde/sip2rtsp:main-c36cb1e@sha256:9bb358c612ae07e3f19d6137d0667c37aa3f572c63a86c6f809043e45c5d8afe`
